Support/Fuego/Pythia/pythia-0.5/packages/fuego/fuego/serialization/c/CPickler.py

```diff
@@ -13372,48 +13258,26 @@ def _sparsity(self, mechanism):
         self._write("{")
         self._indent()
 
-        self._write("int offset;")
         self._write(
-            "amrex::Gpu::DeviceVector<amrex::Real> J_v(%d);"
+            "amrex::GpuArray<amrex::Real,%d> Jac = {0.0};"
             % (nSpecies + 1) ** 2
         )
         self._write(
-            "amrex::Gpu::DeviceVector<amrex::Real> c_v(%d);" % (nSpecies)
-        )
-        self._write("amrex::Real * J_d = J_v.data();")
-        self._write("amrex::Real * c_d = c_v.data();")
-        self._write()
-        self._write("amrex::Real J_h[%d];" % (nSpecies + 1) ** 2)
-        self._write()
-        self._write("amrex::IntVect iv(AMREX_D_DECL(0,0,0));")
-        self._write("amrex::ParallelFor(amrex::Box(iv,iv),")
-        self._write(
-            "    [=] AMREX_GPU_HOST_DEVICE (int /*i*/, int /*j*/, int /*k*/) noexcept {"
+            "amrex::GpuArray<amrex::Real,%d> conc = {0.0};" % (nSpecies)
         )
-        self._write("        for (int k=0; k<%d; k++) {" % (nSpecies))
-        self._write("            c_d[k] = 1.0/ %f ;" % (nSpecies))
-        self._write("        }")
-        self._write("        aJacobian(J_d, c_d, 1500.0, *consP);")
-        self._write("});")
+        self._write("for (int n=0; n<%d; n++) {" % (nSpecies))
+        self._write("    conc[n] = 1.0/ %f ;" % (nSpecies))
+        self._write("}")
+        self._write("aJacobian(&Jac[0], &conc[0], 1500.0, *consP);")
         self._write()
-        self._outdent()
-        self._write("#ifdef AMREX_USE_GPU")
-        self._indent()
-        self._write("amrex::Gpu::dtoh_memcpy(J_h, J_d, sizeof(J_d));")
-        self._outdent()
-        self._write("#else")
-        self._indent()
-        self._write("std::memcpy(&J_h, J_d, sizeof(J_h));")
-        self._outdent()
-        self._write("#endif")
-        self._indent()
+
         self._write("if (base == 1) {")
         self._indent()
         self._write("rowPtr[0] = 1;")
         self._write("int nJdata_tmp = 1;")
         self._write("for (int nc=0; nc<NCELLS; nc++) {")
         self._indent()
-        self._write("offset = nc * %d;" % (nSpecies + 1))
+        self._write("int offset = nc * %d;" % (nSpecies + 1))
```

Comment:
Some of these can probably be `const` but I don't really care.

Reply:
Let's put a pin on that and update next time we have a significant update of the CPickler, to avoid regenerating all the mechs again. PelePhysics repo is starting to get big.